Assets/World/Units/Unit.gd

```diff
@@ -152,6 +153,13 @@ func animate_movement() -> void:
 
 	set_rotation_degree(rotation_index)
 
+func take_damage(damage: int) -> void:
+	if (health - damage) < 0:
+		health = 0
+	else:
+		health -= damage
+	if health == 0:
+		queue_free()
 # =================================================
```

Comment:
```
	set_rotation_degree(rotation_index)

func take_damage(damage: int) -> void:
	if (health - damage) < 0:
		health = 0
	else:
		health -= damage
		
	if health == 0:
		queue_free()
		
# =================================================
# TODO: For later use, hover effect, selection etc.
# =================================================
```
Removed a bit too many lines. Maybe the GitHub single line comments didn't clearly reflect the issues, so I post it here how it should look like in the end.